### PR TITLE
Poll smart contract V0

### PR DESCRIPTION
**Pull request by Suri-Feng**

### CODE REVIEW

File: contracts/Poll.sol [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    event participantLookedUp(string name, address _address);
+    event pollCreated(address organizer, string name, uint dur, bool blind, bool aboutDAO);
+    event voteDone(address voter, bool voted);
+
+    modifier newParticipantCheck(string memory name){
+        require(bytes(name).length > 0, "Participant name is empty");
+        require(bytes(participantName[name]).length == 0, "Participant already registered");
+
+        require(msg.value >= registrationPrice, "Asset not enough to register");
+        require(msg.value == registrationPrice, "Amount sent not equal to the registration price");
```
Comment (astridLi0402) [inline contracts/Poll.sol:66]: What is the registration price for?

File: contracts/Poll.sol [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    {
+        PollEvent storage thisPoll = polls[pollId];
+        require(bytes(thisPoll).length > 0, "Poll not created");
+        require(thisPoll.state == State.VOTING, "The poll has ended");
+        require(bytes(choice).length > 0, "No choice made yet");
+
+        _;
+    }
+
+    function vote(uint pollId, Selection choice)
```
Comment (astridLi0402) [inline contracts/Poll.sol:169]: I feel like we should have a map mapping between participants and their voting status: 
for example, [participant # 1: Jerry; False]: Jerry hasn't voted yet. 
Is that what you're trying to do here?

File: contracts/Poll.sol (hunk 1 above)
Comment (Suri-Feng) [reply contracts/Poll.sol:66]: See the function registerParticipant(). Everyone pays a fee to join the community

File: contracts/Poll.sol (hunk 2 above)
Comment (Suri-Feng) [reply contracts/Poll.sol:169]: If we use the way you suggested, we can only record Jerry's vote for one poll, but we want to record down his vote for every poll, that's why I store the voting information in ` address[] voted // An array to store who has already voted` and  `Selection[] votedChoices; // An array to store voter's choices `

**Pull request merged**